weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/context.tsx

```diff
@@ -117,3 +117,51 @@ export const useProjectHasTraceServerCalls = (
 
   return (calls.result ?? []).length > 0;
 };
+
+/**
+ * Returns true if the client can connect to trace server and the project has
+ * objects.
+ */
+export const useProjectHasTraceServerObjects = (
+  entity: string,
+  project: string
+) => {
+  const hasTraceServer = useHasTraceServerClientContext();
+  const objs = tsWFDataModelHooks.useRootObjectVersions(
+    entity,
+    project,
+    {},
+    1,
+    {
+      skip: !hasTraceServer,
+    }
+  );
+
+  return (objs.result ?? []).length > 0;
+};
+
+/**
+ * Returns true if the client can connect to trace server and the project has
+ * objects or calls.
+ */
+export const useProjectHasTraceServerData = (
```

Comment:
For the nav bar stuff I'm doing I need a version of this that returns the loading state in addition to the boolean value, otherwise the 0 result while loading will cause an annoying flash of the wrong buttons.

Reply:
Ill add loading state